daml-lf/validation/src/test/scala/com/digitalasset/daml/lf/validation/TypingSpec.scala

```diff
@@ -767,20 +792,56 @@ class TypingSpec extends AnyWordSpec with TableDrivenPropertyChecks with Matcher
           { case _: ETypeMismatch => },
         E"Λ (σ : ⋆).λ (e₁: ContractId σ) (e₂: List Party) (e₃: Int64) → ⸨ exercise @Mod:T Ch e₁ e₂ e₃ ⸩" -> //
           { case _: ETypeMismatch => },
-        // FecthByKey & lookupByKey
-        E"""⸨ fetch_by_key @Mod:U "Bob" ⸩""" -> //
+        // This verifies that template choice cannot be exercise by interface
+        E"λ (e₁: ContractId Mod:I) (e₂: List Party) (e₃: Int64) → ⸨ exercise @Mod:I ChTmpl e₁ e₂ e₃ ⸩" -> {
+          case EUnknownDefinition(
+                _,
+                LookupError(Reference.Template(_), Reference.TemplateChoice(_, _)),
+              ) =>
+            // We double check that Ti implements I and Ti has a choice ChTmpl
+            val TTyCon(conI) = t"Mod:I"
+            val TTyCon(conTi) = t"Mod:Ti"
+            assert(env.interface.lookupTemplateImplements(conI, conTi).isRight)
+            assert(env.interface.lookupTemplateChoice(conTi, n"ChTmpl").isRight)
+        },
+        // UpdExerciseInterface
+        E"λ (e₁: ContractId Mod:U) (e₂: List Party) (e₃: Int64) → ⸨ exercise_by_interface @Mod:U ChIface e₁ e₂ e₃ ⸩" -> //
           {
             case EUnknownDefinition(
                   _,
-                  LookupError(Reference.Template(_), Reference.TemplateKey(_)),
+                  LookupError(Reference.Interface(_), Reference.InterfaceChoice(_, _)),
                 ) =>
           },
-        E"""⸨ fetch_by_key @Mod:T "Bob" ⸩""" -> //
+        E"λ (e₁: ContractId Mod:I) (e₂: List Party) (e₃: Int64) → ⸨ exercise_by_interface @Mod:I Not e₁ e₂ e₃ ⸩" -> //
+          {
+            case EUnknownDefinition(
+                  _,
+                  LookupError(Reference.TemplateChoice(_, _), Reference.InterfaceChoice(_, _)),
+                ) =>
+          },
+        E"Λ (σ : ⋆).λ (e₁: ContractId Mod:I) (e₂: List Party) (e₃: σ) → ⸨ exercise_by_interface @Mod:T ChIface e₁ e₂ e₃ ⸩" -> //
```

Comment:
same question here, it looks like you’re trying to pass actors for some reason.

Reply:
fixed